components/permissions/contexts/brave_ethereum_permission_context.cc

```diff
@@ -0,0 +1,85 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/permissions/contexts/brave_ethereum_permission_context.h"
+
+#include <utility>
+
+#include "base/no_destructor.h"
+#include "brave/components/brave_wallet/browser/ethereum_permission_utils.h"
+#include "components/content_settings/core/common/content_settings_types.h"
+#include "components/permissions/permission_request_id.h"
+#include "content/public/browser/web_contents.h"
+#include "third_party/blink/public/mojom/permissions_policy/permissions_policy_feature.mojom.h"
+#include "third_party/re2/src/re2/re2.h"
+
+namespace {
+constexpr char kAddrPattern[] = "addr%3D(0x[[:xdigit:]]{40})";
```

Comment:
To avoid this being used without verify the input as expected format, I decided to wrap the extraction logic into ParseRequestingOrigin rather than exposing addr-extractor function, tests are updated too.
Fixed in 7edcc74.